baremaps-postgres/src/main/java/com/baremaps/postgres/jdbi/FeatureMapper.java

```diff
@@ -0,0 +1,59 @@
+package com.baremaps.postgres.jdbi;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+
+public class FeatureMapper implements RowMapper<Feature> {
+
+  private final GeometryColumnMapper mapper = new GeometryColumnMapper();
+
+  @Override
+  public Feature map(ResultSet rs, StatementContext ctx) throws SQLException {
+    var feature = new Feature();
+    feature.setProperties(new HashMap<>());
+
+    var metadata = ctx.getStatement().getMetaData();
+    var columnCount = metadata.getColumnCount();
+    for (int i = 1; i <= columnCount; i++) {
+      var columnName = metadata.getColumnName(i);
+      switch (columnName) {
+        case "id":
+        case "fid":
```

Comment:
What about if the column name is set with "Fid".?
I mean about 
```CREATE TABLE testtable (id int, "Fid" int);```

Reply:
Some importers for shapefiles create 'fid' and 'gid' columns for the primary key. This is an intent to make the RowMapper a bit smarter if these fields exist.

https://support.esri.com/en/technical-article/000010834

Reply:
Ok. It's a workaround for a specific use case.  A solution will be to check if there is a primary key on the table and use it as fid. If no primary key exists we can use the internal OID. 
Because the mapper iterate row by row, not sure if it's possible to get some table constraints from the query.  
On the other hand it works only if the request is simple. If the query plays with several tables it will be harder. ;(
So to be generic I'm favour to get only the properties in a hashmap without specific workaround. Note this is just a point of view ; -)

Reply:
I may be missing something, but I don't think that the metadata of a select statement contains information about the primary key (`ctx.getStatement().getMetaData()`).

The generic approach you describe would be ideal. A possibility may be to generate the select queries from the metadata of the table (to identify the primary key), but I'm not sure if this level of introspection can be implemented with the abstractions provided by JDBI.

Reply:
You right. 
PK column is defined within database system table.